libs/area_reporting.py

```diff
@@ -0,0 +1,72 @@
+import os
+import time
+import logging
+import csv
+from datetime import date
+from collections import deque
+from .utils.mailing import MailService
+from .notifications.slack_notifications import SlackService
+
+logger = logging.getLogger(__name__)
+
+
+class AreaReporting:
+
+    def __init__(self, config, area):
+        self.processing_alerts = False
+        self.config = config
+        self.area = area
+
+        self.occupancy_sleep_time_interval = float(self.config.get_section_dict("App")["OccupancyAlertsTimeout"])
+        self.log_dir = self.config.get_section_dict("Logger")["LogDirectory"]
+        self.idle_time = float(self.config.get_section_dict('Logger')['TimeInterval']) + 0.1
```

Comment:
If the idea is to process this for every frame processed from the camera then here you will be missing a few frames every minute. I think the ideal would be to sleep For TimeInterval seconds. Or maybe we don't want to check this as often. Maybe it is enough to check it every few seconds?

Reply:
Yeah, I think you're right. I thought about putting a small delay so that if a processor takes longer then it would be included, but this would cause the processing to be even more unpredictable